### PR TITLE
Manage logging in PPE

### LINKED ISSUE #172: Create internal debug tools
We need some kind of logging mechanism for debug

When loading snapshots or verifying trigger checks having a 
`PPE_LOG_LEVEL` environment variable that allows us to obtain DEBUG information 
(snapshot not found, snapshot invalid, etc.) will be very useful.

### PR DESCRIPTION
**Pull request by emcastillo**
Fixes #172

### CODE REVIEW

**Comment by kmaehashi** [comment]
Could you avoid using or configuring the root (default) logger? Please create a dedicated logger instance for PPE and use it from modules in PPE.

File: pytorch_pfn_extras/__init__.py [hunk 1]
```diff
@@ -15,3 +17,27 @@
 from pytorch_pfn_extras._tensor import from_numpy_dtype  # NOQA
 
 from pytorch_pfn_extras._version import __version__  # NOQA
+
+
+def configureLogging(filename=None, level=logging.ERROR):
```
Comment (kmaehashi) [inline pytorch_pfn_extras/__init__.py:22]:
```suggestion
def configureLogging(filename=None, level='ERROR'):
```

**Comment by emcastillo** [comment]
I think this can be reviewed

File: pytorch_pfn_extras/logging.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import logging
+import os
+
+_logger_name = 'ppe'
+_logger = None
+
+
+def configure_logging(filename=None, level='ERROR'):
```
Comment (kmaehashi) [inline pytorch_pfn_extras/logging.py:8]: This should be a private method and without arguments as users cannot change the filename / level after importing PPE.

```suggestion
def _configure_logging():
```

File: pytorch_pfn_extras/__init__.py [hunk 3]
```diff
@@ -1,3 +1,7 @@
+# Configure the logging before instantiating anything else
+from pytorch_pfn_extras import logging  # NOQA
+logging.configure_logging()
```
Comment (kmaehashi) [inline pytorch_pfn_extras/__init__.py:3]:
```suggestion
logging._configure_logging()
```

File: pytorch_pfn_extras/logging.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    global _logger
+    filename = os.environ.get('PPE_LOG_FILENAME', filename)
```
Comment (kmaehashi) [inline pytorch_pfn_extras/logging.py:10]:
```suggestion
    filename = os.environ.get('PPE_LOG_FILENAME', None)
```

File: pytorch_pfn_extras/logging.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    if filename is None:
+        handler = logging.StreamHandler()
+    else:
+        handler = logging.FileHandler(filename)
+    # To dynamically change the level if needed
+    # basicConfig does not allow to change the level right after
+    _logger = logging.getLogger(_logger_name)
+    level = os.environ.get('PPE_LOG_LEVEL', level)
```
Comment (kmaehashi) [inline pytorch_pfn_extras/logging.py:18]:
```suggestion
    level = os.environ.get('PPE_LOG_LEVEL', 'ERROR')
```

File: pytorch_pfn_extras/logging.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    for lvl in (logging.DEBUG, logging.INFO,
+                logging.WARNING, logging.ERROR, logging.CRITICAL):
+        if logging.getLevelName(lvl) == level:
+            _logger.setLevel(lvl)
+            break
+    else:
+        _logger.warning('invalid PPE_LOG_LEVEL (%s); using INFO', level)
+        _logger.setLevel(logging.INFO)
```
Comment (kmaehashi) [inline pytorch_pfn_extras/logging.py:26]:
```suggestion
        _logger.setLevel(logging.INFO)
        _logger.warning(f'invalid PPE_LOG_LEVEL ({level}); using INFO')
```

File: pytorch_pfn_extras/training/extensions/_snapshot.py [hunk 7]
```diff
@@ -3,9 +3,14 @@
 import torch
 import torch.distributed
 
+from pytorch_pfn_extras import logging
 from pytorch_pfn_extras.training import extension
 
 
+logger = logging.get_logger()
+print(logger)
```
Comment (kmaehashi) [inline pytorch_pfn_extras/training/extensions/_snapshot.py:11]:
```suggestion
```

**Review by kmaehashi — CHANGES_REQUESTED** [review]

**Comment by emcastillo** [comment]
Fixes applied :)

**Comment by kmaehashi** [comment]
Thanks! Could you check this? https://github.com/emcastillo/pytorch-pfn-extras/pull/1

**Review by kmaehashi — APPROVED** [review]

**Pull request merged**